### PR TITLE
Finish up preparation for v3 of site

### PR DESCRIPTION
**Pull request by tristan957**
Signed-off-by: Tristan Partin <tpartin@micron.com>

### CODE REVIEW

**Comment by tristan957** [comment]
```
meson setup build
ninja -C build
ninja -C build deploy
ninja -C build serve
```

Still defaults to 2.x. Not sure how to change that.

**Comment by tristan957** [comment]
I really really really don't like how GitHub Pages works.

File: v2/mkdocs.yml [hunk 1]
```diff
@@ -24,9 +24,6 @@ extra_css:
 # Theme configuration
 theme:
   name: material
-  # custom_dir extends the material theme to add an announcement bar
-  # indicating that HSE 2.x is no longer actively maintained
-  custom_dir: overrides
```
Comment (smoyerx) [inline v2/mkdocs.yml]: This is required to display the obsolete message.

**Comment by smoyerx** [comment]
> ```
> meson setup build
> ninja -C build
> ninja -C build deploy
> ninja -C build serve
> ```
> 
> Still defaults to 2.x. Not sure how to change that.

mike set-default 3.x

**Comment by smoyerx** [comment]
> I really really really don't like how GitHub Pages works.

Don't know why, it's easy, and nothing to manage.  But that's outside the scope of this PR....

**Comment by smoyerx** [comment]
What is replacing the requirements.txt?

**Comment by tristan957** [comment]
> What is replacing the requirements.txt?

Why does something need to replace it?

**Comment by smoyerx** [comment]
> > What is replacing the requirements.txt?
> 
> Why does something need to replace it?

Sure, the alternative is to specify what to install in README.md.

File: v3/docs/api/c/meson.build [hunk 2]
```diff
@@ -40,6 +36,6 @@ generated_docs += custom_target(
         meson.current_source_dir() / '../.doxybook/templates'
     ],
     depends: [
-        c_doxygen,
+        c_docs,
```
Comment (bhaveshvasandani) [inline v3/docs/api/c/meson.build:39]: The input option points to subprojects/hse-r2.1.1? Is that something we are waiting to tag the r3.0.0?

Comment (tristan957) [reply v3/docs/api/c/meson.build:39]: Whoops. No, I am just an idiot! Great catch

File: subprojects/hse-3.wrap [hunk 3]
```diff
@@ -0,0 +1,9 @@
+[wrap-git]
+url = https://github.com/hse-project/hse.git
+revision = master
```
Comment (bhaveshvasandani) [inline subprojects/hse-3.wrap]: revision = v3.0?

Comment (tristan957) [reply subprojects/hse-3.wrap]: There we no tag at the time. Is there a tag now?

File: v3/docs/gs/params.md [hunk 4]
```diff
@@ -7,9 +7,8 @@ or *runtime* parameters, which apply each time an application
 initializes the HSE library or opens a KVDB or KVS.
 
 In addition to the detailed descriptions below, you can find parameter
-usage examples in the [API reference](../api/c/Pages/examples.md),
-[CLI reference](cli.md), and HSE
-[sample programs](https://github.com/hse-project/hse/tree/master/samples).
+usage in the [CLI reference](cli.md) and HSE
+[sample programs](https://github.com/hse-project/hse/tree/v3.0/samples).
```
Comment (smoyerx) [inline v3/docs/gs/params.md]: Why is it ".../tree/v3.0/samples" here and ".../tree/r2/samples" in v2/docs/params.md?

**Review by smoyerx — COMMENTED** [review]
Can build docs at this point.  Nits: 1) "ninja -C build serve" has versions 2.x and 1.x rather than 3.x and 2.x, and 2) "ninja -C build deploy" does not issue a "mike set-default 3.x", however this can be done manually before pushing the gh-pages branch, along with "mike delete 1.x" to remove that version.

File: v3/docs/gs/params.md (hunk 4 above)
Comment (tristan957) [reply v3/docs/gs/params.md]: Because hse 3.0 hasn't been tagged yet, but I guess it is just a link so it doesn't matter

**Comment by tristan957** [comment]
> Can build docs at this point. Nits: 1) "ninja -C build serve" has versions 2.x and 1.x rather than 3.x and 2.x, and 2) "ninja -C build deploy" does not issue a "mike set-default 3.x", however this can be done manually before pushing the gh-pages branch, along with "mike delete 1.x" to remove that version.

Did you run deploy before serve? Try deleting the gh-pages branch before deploying & serving. There is nothing I can change with regard to what is in your local gh-pages branch.

This is why I don't like this model of this special branch that GitHub Pages has. People don't expect things to actually be the way they should be.

**Comment by smoyerx** [comment]
I forgot to run deploy before serve, my bad.  Now when I serve I see all three versions (1.x, 2.x, 3.x) with 2.x being the default.  This is actually what I would expect given that 1) we need to manually remove 1.x via "mike delete 1.x", and 2) after "ninja -C build deploy" we need to manually run "mike set-default 3.x", which only needs to be run once until there is another major version.

Update: Just confirmed all this is as expected.  I ran "mike delete 1.x" and "mike set-default 3.x" and then "ninja -C build serve" showed just 2.x and 3.x with 3.x the default as expected.  Then ran "ninja -C build deploy" followed by "ninja -C build serve" and again got expected results (2.x and 3.x with 3.x the default).

So it all looks good.  Let's just change the example link from the comment above and ship it.

File: v3/docs/gs/params.md (hunk 4 above)
Comment (smoyerx) [reply v3/docs/gs/params.md]: Right, let's make them consistent.  Before this update it just pointed to latest for all doc versions, but actually that wasn't correct given changes between major versions.

**Review by smoyerx — APPROVED** [review]

**Comment by tristan957** [comment]
I think we should actually keep this open until HSE 3.0 is tagged. That way I can change the hse-3 wrap file.

**Comment by smoyerx** [comment]
> I think we should actually keep this open until HSE 3.0 is tagged. That way I can change the hse-3 wrap file.

Makes sense.

File: subprojects/hse-3.wrap (hunk 3 above)
Comment (bhaveshvasandani) [reply subprojects/hse-3.wrap]: I mean there is the branch v3.0...

Comment (tristan957) [reply subprojects/hse-3.wrap]: oh whoops, when this PR was created there was no branch :)

Comment (tristan957) [reply subprojects/hse-3.wrap]: any who, waiting for the release now

**Review by bhaveshvasandani — APPROVED** [review]

**Pull request merged**